src/components/form/templates/common.ts

```diff
@@ -50,28 +52,53 @@ export function findTitle(data: any, fieldSchema: any, formSchema: any) {
         return findSchemaTitle(data, fieldSchema);
     }
 
-    const firstEntry = findFirstEntry(data);
+    const subSchema = findSubSchema(fieldSchema, formSchema);
+
+    data = sortDataByProperties(data, subSchema.properties);
+
+    const firstEntry = findFirstEntry(data, subSchema);
     if (!firstEntry) {
         return null;
     }
 
     const [key, value] = firstEntry;
-    const subSchema = findSubSchema(fieldSchema, key, formSchema);
+    return findTitle(value, subSchema.properties[key], formSchema);
+}
 
-    return findTitle(value, subSchema, formSchema);
+function sortDataByProperties(data: any, properties: object) {
+    if (!properties || isEmpty(properties)) {
+        return data;
+    }
+
+    const newData = {};
+    Object.keys(properties).forEach(key => (newData[key] = data[key]));
+    return newData;
 }
 
-function findFirstEntry(data: any) {
+function findFirstEntry(data: any, subSchema: any) {
     const entries = [
         ['title', data.title],
         ['name', data.name],
         ...Object.entries(data),
     ];
 
+    if (subSchema.required) {
+        const requiredEntry = getRequiredEntry(data, subSchema);
+        const requiredEntryIndex = 2;
+        entries[requiredEntryIndex] = requiredEntry;
```

Comment:
This will overwrite the first property from data :scream: 
What do you think about adding a new method called `getRequiredEntries` and write the code above like this?
```typescript
const entries = [
    ['title', data.title],
    ['name', data.name],
    ...getRequiredEntries(),
    ...Object.entries(data),
    ];
```